Hippo.Web/ClientApp/src/components/Order/Details.tsx

```diff
@@ -0,0 +1,685 @@
+  const [isClusterAdmin, setIsClusterAdmin] = useState(null);
+  const [notification, setNotification] = usePromiseNotification();
+  const [updateStatusModel, setUpdateStatusModel] =
+    useState<UpdateOrderStatusModel | null>(null);
+  const adminEditableStatuses = ["Processing", "Active"];
+  const sponsorEditableStatuses = ["Created"];
+
+  useEffect(() => {
+    setIsClusterAdmin(isClusterAdminForCluster());
+  }, [isClusterAdmin, isClusterAdminForCluster]);
+
+  const calculateBalanceRemaining = (data: any) => {
+    const balanceRemaining = parseFloat(data.balanceRemaining);
+    setBalanceRemaining(balanceRemaining);
+    const balancePending = data?.payments
+      .filter(
+        (payment) =>
+          payment.status !== "Completed" && payment.status !== "Cancelled",
+      )
+      .reduce((acc, payment) => acc + parseFloat(payment.amount), 0);
+    setBalancePending(balancePending);
+  };
+
+  useEffect(() => {
+    const fetchOrder = async () => {
+      const response = await authenticatedFetch(
+        `/api/${cluster}/order/get/${orderId}`,
+      );
+
+      if (response.ok) {
+        const data = await response.json();
+        console.log(data);
+        setOrder(data);
+        calculateBalanceRemaining(data);
+      } else {
+        alert("Error fetching order");
+      }
+    };
+
+    fetchOrder();
+  }, [cluster, orderId]);
+
+  useEffect(() => {
+    if (order) {
+      // switch statement for data.status
+      switch (order.status) {
+        case "Created":
+          setUpdateStatusModel({
+            currentStatus: order.status,
+            newStatus: "Submitted",
+          });
+          break;
+        case "Submitted":
+          setUpdateStatusModel({
+            currentStatus: order.status,
+            newStatus: "Processing",
+          });
+          break;
+        case "Processing":
+          setUpdateStatusModel({
+            currentStatus: order.status,
+            newStatus: "Active",
+          });
+          break;
+        default:
+          setUpdateStatusModel({
+            currentStatus: order.status,
+            newStatus: order.status,
+          });
+      }
+    }
+  }, [order]);
+
+  const historyColumnHelper = createColumnHelper<HistoryModel>();
+
+  const historyColumns = [
+    historyColumnHelper.accessor("actedDate", {
+      header: "Date",
+      id: "actedDate",
+      cell: (value) => (
+        <span>{convertToPacificTime(value.row.original.actedDate)}</span>
+      ),
+    }),
+    historyColumnHelper.accessor("actedBy", {
+      header: "Actor",
+      id: "actedBy",
+      cell: (value) => (
+        <>
+          {value.row.original.actedBy && (
+            <>
+              {value.row.original.actedBy.firstName}{" "}
+              {value.row.original.actedBy.lastName} (
+              {value.row.original.actedBy.email})
+            </>
+          )}
+          {!value.row.original.actedBy && <>System</>}
```

Comment:
clearer if you use a tertiary (`actedBy ? (name) : system`) instead of doing multiple checks against the actedBy value.